tests/scripts/thread-cert/Cert_7_1_01_BorderRouterAsLeader.py

```diff
@@ -106,12 +118,58 @@ def test(self):
             if addr[0:10] == '2001:2:0:1' or addr[0:10] == '2001:2:0:2':
                 self.assertTrue(self.nodes[LEADER].ping(addr))
 
-        addrs = self.nodes[ED1].get_addrs()
+        addrs = self.nodes[MED1].get_addrs()
         self.assertTrue(any('2001:2:0:1' in addr[0:10] for addr in addrs))
         self.assertTrue(any('2001:2:0:2' in addr[0:10] for addr in addrs))
         for addr in addrs:
             if addr[0:10] == '2001:2:0:1' or addr[0:10] == '2001:2:0:2':
                 self.assertTrue(self.nodes[LEADER].ping(addr))
 
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        med1_messages = self.simulator.get_messages_sent_by(MED1)
+        sed1_messages = self.simulator.get_messages_sent_by(SED1)
+
+        # Step 1 - DUT sends MLE Advertisements
+        msg = leader_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+
+        # Step 2 - DUT creates network data
+        msg = leader_messages.next_mle_message(mle.CommandType.DATA_RESPONSE)
+        check_data_response(msg, network_data_opt=CheckType.CONTAIN,
+            prefixes=[ b'2001000200000001', b'2001000200000002' ])
+
+        # Step 4 - DUT sends a MLE Child ID Response to Router1
+        msg = leader_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
+        # Only check the network_data in the child id response,
+        # pass a network_data check function to check the detail
+        check_child_id_response(msg, network_data=CheckType.CONTAIN, \
+            network_data_detail_check_func=functools.partial(check_network_data_tlv, min_prefix_count=2, prefix_check_func=check_prefix))
```

Comment:
It looks to me a little complex to specify the check_prefix entry here. 
Could we simply pass the essential parameters and check requirements in the test script here and leave such concrete correlations in the common utility component (`command.py`)?